src/lib/y2storage/encryption.rb

```diff
@@ -322,16 +322,63 @@ def save_crypttab_name(devicegraph, entry)
 
         return unless device
 
-        if device.encrypted?
-          device.encryption.crypttab_name = entry.name
-        elsif entry.swap?
-          # Right now, a plain encryption device is created for swap only. Moreover, random password
-          # (urandom) is always considered, even when the crypttab file indicates another key file.
+        if create_swap_encryption_for_entry?(device, entry)
           device.remove_descendants
-          device.encrypt(dm_name: entry.name, method: :random_swap)
+          device.encrypt(dm_name: entry.name, method: encryption_method_for_entry(entry))
+        elsif device.encrypted?
+          device.encryption.crypttab_name = entry.name
         end
       end
 
+      # Whether the encrytion layer should be created for a swap device according to the given crypttab
+      # entry
+      #
+      # @param device [BlkDevice]
+      # @param entry [SimpleEtcCrypttabEntry]
+      #
+      # @return [Boolean]
+      def create_swap_encryption_for_entry?(device, entry)
+        encryption_method = encryption_method_for_entry(entry)
+
+        return false unless encryption_method_for_swap?(encryption_method)
```

Comment:
Couldn't this be simply?

```
return false unless encryption_method&.only_for_swap?
```

Sounds simpler than defining an additional method.

Reply:
Right now, we could because `#only_for_swap?` means basically the same as `#encryption_method_for_swap?`, but I would prefer to have them as separate methods.

Reply:
I have tried to improve the method description.